### ISSUE TITLE
VCH should not allow ContainerVMs to run if their vulnerability degree exceeds the project threshold

### ISSUE THREAD

**Issue by kreamyx** [vmware/vic#4562]

**Comment by kreamyx** [comment]
Based on a conversation with @hickeng and @akshayl, the bulk of the logic of this requirement will be at the Harbor side

**Comment by akshayl** [comment]
You should check with the Harbor @reasonerjt and Admiral tech leads to confirm the architecture proposal. I have no opinion on where the logic should reside.

**Comment by hickeng** [comment]
I have real concerns about this - the user story I can get behind, but I think enforcing it conceptually at `docker run` is flawed.
My concern is focused around event sequences such as:

1. deploy app into production - images are pulled to some docker hosts.
2. new vulnerabilities are found in an image that's part of the app and the project threshold is now violated.

In a very, very static environment this isn't necessarily a problem, however container orchestration isn't engineered in that way and it doesn't account for outages:

a. host goes down - containers should come up on another host but one does not because it violates threshold - **production app is now broken**
b. orchestration rebalances containers - cannot start one container on a new host because it violates threshold - **production app is now broken**

This needs some very careful planning for possible scenarios and desired behaviours before anything automatic is implemented.

**Comment by karthik-narayan** [comment]
@hickeng I agree that these are very valid concerns and most customers are guaranteed to hit one or more of these. 

The question is, can these be addressed in time for 1.2?

**Comment by akshayl** [comment]
What are some alternative ways to enforce the requirement? Is it possible to differentiate between "first run" and "rerun" due to rebalancing / reorchestration?

**Comment by akshayl** [comment]
The PRD addresses another related uses case i.e. "Handle Vulnerabilities in Running Containers". Perhaps we could use the behavior specified there for the scenarios you outlined?

**Comment by akshayl** [comment]
I have updated the requirements document to account for these corner cases.

**Comment by mdubya66** [comment]
Removing from 1.2
